app/main.py

```diff
@@ -2,5 +2,14 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    saved_runs = {}
+
+    def wrapper(*args) -> (int, float):
+        if args not in saved_runs.keys():
+            print("Calculating new result")
+            saved_runs[args] = func(*args)
+            return saved_runs[args]
+        else:
+            print("Getting from cache")
+            return saved_runs[args]
```

Comment:
It's recommended to do a direct condition first(not always, but when you have if and else, then you can simplify it) and then else because it'll be easier to read like
```
# this is easier to read
if something:
    pass
# else means "not something"
else:
    pass

# than this one
if not something:
    pass
# else means "**not** **not** something"
else:
    pass
```

And in general you don't need else statement here because "if" statement returns value, thus it'll not reach the code after this ;)